SDL Compliance Completion

AgneLukoseviciute opened the pull request:
Adds two forked folly files in order to re-enable C4244 and C4267 warning suppressions. I opened a [PR in folly](https://github.com/facebook/folly/pull/1565) to address this from there, and will add an issue to track updating folly and removing these files. 
Closes #6918. 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7659)

asklar commented on vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:
```diff
@@ -101,7 +101,7 @@
       <PrecompiledHeaderOutputFile>$(IntDir)pch.pch</PrecompiledHeaderOutputFile>
       <ForcedIncludeFiles>pch.h</ForcedIncludeFiles>
-      <WarningLevel>Level4</WarningLevel>
+      <WarningLevel>Level3</WarningLevel>
```
Why is this needed? Seems like this will disable a bunch of warnings

AgneLukoseviciute replied:
Not needed, I'll change that back.

asklar commented on vnext/Folly/TEMP_UntilFollyUpdate/lang/ToAscii.h:
```diff
@@ -0,0 +1,420 @@
+#if FOLLY_X64
+  return static_cast<int>(__lzcnt64(v));
+#else
+  __assume(0);
```
Do x86 builds use FOLLY_X64?

AgneLukoseviciute replied:
From working on this it appears to be, but I'm currently trying to figure out how it works exactly for the fix in folly b/c the return type of this method changed to auto but moving the cast to the caller on line 227 only works in x64. x86 is complaining about trying to convert void to int.

asklar reviewed (changes requested):
Left some questions. 

Meta point (this one shouldn't block this PR), it might be better to use patches instead of forked files to be able to PR the individual forked change rather than an entire file that has maybe one line different. Thoughts?

AgneLukoseviciute commented:
> Left some questions.
> 
> Meta point (this one shouldn't block this PR), it might be better to use patches instead of forked files to be able to PR the individual forked change rather than an entire file that has maybe one line different. Thoughts?

@asklar That sounds like a better approach, especially with the minimal changes. I can update this to use patches instead.

acoates-ms commented:
/azp run

asklar commented:
[talked offline, we still need to figure out how x86 should/will work]

AgneLukoseviciute commented:
@asklar, I updated to_ascii_port_clzll() method to call builtin_clzll() on x86. builtin_clzll() will then call __builtin_clz() which is supported only in VS 16.3-16.8 but code added [in folly](https://github.com/facebook/folly/pull/1450/files) handles the logic for calling polyfill that uses _BitScanReverse when __builtin_clz is not available

With the forked files it's hard to see the diffs; for reference the only changes I've made to these files from what we were using before are in ToAscii.h lines 59-63(adding check for x86), 64/96(updates to match changes upstream - return type changed to auto from int and added return statement in line 69), 84(added cast), & 233(added cast). in Dynamic.cpp the only change is an added cast on line 300.
